### PR TITLE
Merge InternalContext with React-hook-form context

### PR DESCRIPTION
**Pull request by manni497**

### CODE REVIEW

File: src/lib/context/FormContext.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  disabled: boolean;
+}
+
+export interface InternalFormContextProviderProps<T extends FieldValues> extends Pick<FormContextProps<T>, "requiredFields" | "disabled"> {
+  children: React.ReactNode;
+  formMethods: UseFormReturn<T, unknown>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const FormContext = createContext<FormContextProps<any>>({} as FormContextProps<any>);
```
Comment (neotrow) [inline src/lib/context/FormContext.tsx:15]:
```suggestion
export const FormContext = createContext<FormContextProps<any> | null>(null);
```

File: src/lib/context/FormContext.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      }}
+    >
+      {children}
+    </FormContext.Provider>
+  );
+};
+
+export const useFormContext = <T extends FieldValues>() => {
+  const context = useContext(FormContext);
+
```
Comment (neotrow) [inline src/lib/context/FormContext.tsx:35]: Would be good to add this:

```
  if (!context) {
    throw new Error("useFormContext must be used within a FormContextProvider");
  }
```

File: src/lib/context/FormContext.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export interface InternalFormContextProviderProps<T extends FieldValues> extends Pick<FormContextProps<T>, "requiredFields" | "disabled"> {
+  children: React.ReactNode;
+  formMethods: UseFormReturn<T, unknown>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const FormContext = createContext<FormContextProps<any>>({} as FormContextProps<any>);
+
+export const InternalFormProvider = <T extends FieldValues>(props: InternalFormContextProviderProps<T>) => {
```
Comment (neotrow) [inline src/lib/context/FormContext.tsx:17]: This is unused and can be removed.

Comment (neotrow) [reply src/lib/context/FormContext.tsx:17]: Than you can also remove `InternalFormContextProviderProps`

**Review by neotrow — CHANGES_REQUESTED** [review]
Thanks for the PR, a few small changes but great otherwise :-)

File: src/lib/context/FormContext.tsx (hunk 2 above)
Comment (manni497) [reply src/lib/context/FormContext.tsx]: Added :D

File: src/lib/context/FormContext.tsx (hunk 3 above)
Comment (manni497) [reply src/lib/context/FormContext.tsx]: @neotrow I'm a stupid,  fixed :D

File: src/lib/context/FormContext.tsx (hunk 1 above)
Comment (manni497) [reply src/lib/context/FormContext.tsx]: @neotrow Done :D

**Review by neotrow — APPROVED** [review]

**Review by neotrow — CHANGES_REQUESTED** [review]
thanks for the changes, sorry forgot in my last review:

Can you please add a section in the Readme? Mainly explaining that the direct `useFormContext` from the `react-hook-form` library won't work and that the one from our library has to be used :-) And maybe also mention that we expose some other stuff in our context.

Many thanks!

**Review by neotrow — APPROVED** [review]

**Pull request merged**